Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -517,11 +517,42 @@ public override void Update()
             if (!IsSetupValid) { return; }
 
             UpdatePointers();
+
+            if (gazeProviderPointingData?.Pointer != null)
+            {
+                UpdateGazeProvider();
+            }
             UpdateFocusedObjects();
 
             PrimaryPointer = primaryPointerSelector?.Update();
         }
 
+        /// <summary>
+        /// Update the hit information for the gaze provider.
+        /// 
+        /// For the sake of StabilizationPlaneModifier and potentially other 
+        /// components that care where the user's looking, we need to do a gaze 
+        /// raycast even if gaze isn't used for focus.
+        /// Assumes gaze provider is not null;
+        /// </summary>
+        private void UpdateGazeProvider()
+        {
+            // We only update if interaction is not enabled because if interaction is enabled 
+            // for this pointer, we have already updated hit data
+            if (!gazeProviderPointingData.Pointer.IsInteractionEnabled)
+            {
+                var raycastProvider = InputSystem.RaycastProvider;
+                // Should we just use a local variable instead of this member?
+                // Unclear why we use the member everywhere...maybe to reduce allocaionts
+                hitResult3d.Clear();
+                LayerMask[] prioritizedLayerMasks = (gazeProviderPointingData.Pointer.PrioritizedLayerMasksOverride ?? FocusLayerMasks);
+
+                QueryScene(gazeProviderPointingData.Pointer, raycastProvider, prioritizedLayerMasks,
+                    hitResult3d, maxQuerySceneResults);
+                ((GazeProvider)InputSystem.GazeProvider).UpdateGazeInfoFromHit(hitResult3d.raycastHit);
```

Comment:
>(GazeProvider) [](start = 17, length = 14)

This has an interesting side effect of preventing other people from implementing their own gaze provider directly using the IMixedRealityGazeProvider interface.

I think that, truth be told, it's really hard to implement the gaze provider correctly without having everything inside GazeProvider.cs, so the value of the interface itself isn't high, but if we are going to do this we should do this consciously and make this tradeoff explicitly.

Reply:
I could add another method to the GazeProvider interface :-)

But honestly I kind of feel like GazeProvider shouldn't be an interface, for this reason:

> I think that, truth be told, it's really hard to implement the gaze provider correctly without having everything inside GazeProvider.cs

Reply:
In the current mrtk, FocusProvider and GazeProvider are quite closely coupled: FocusProvider basically updates all the data for gaze provider. I actually don't think the architects intended this. If you look at the IGazeProvider interface, all its fields are only getters, meaning I think architects intended GazeProvider to update itself. It would actually be pretty easy to see a system where this would work -- in gaze provider's update loop, GazeProvider just does a raycast and updates its fields. But the current code instead has FocusProvider updating GazeProvider. 

I'd rather not add to the interface, since I think its design is not so bad -- to encourage people to have the gaze provider update itself. It's just the specific implementation we have doesn't do that.

Reply:
I was about to write this same comment! I think I'm mostly agreed with the conclusion here, where the more common data provider to pointer relationship seems more muddled with the gaze provider than any of the hand / controller pointers. On initial thought, I'm not sure about the GazeProvider doing its own raycast (to try to get it to follow the pattern of the other providers / pointers as much as possible), but its possible we'll need to special case it.